src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -164,6 +168,21 @@ public ResponseEntity<DatasetSummaryModel> retrieveDatasetSummary(UUID id) {
     return ResponseEntity.ok(datasetService.retrieveDatasetSummary(id));
   }
 
+  @Override
+  public ResponseEntity<DatasetModel> updateDatasetSnapshotBuilderSettings(
+      UUID id, SnapshotBuilderSettings settings) {
+    AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    iamService.verifyAuthorization(
+        userRequest,
+        IamResourceType.DATASET,
+        id.toString(),
+        IamAction.UPDATE_SNAPSHOT_BUILDER_SETTINGS);
```

Comment:
For the time being I think we should have a new action - we only want admins to do this right now. I think maybe eventually it could be rolled into that.